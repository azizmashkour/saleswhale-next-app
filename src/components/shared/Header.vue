<template>
  <section id="header">
    <div class="contact-us">
      <a href="#">OUR COVID-19 RESPONSE</a>
    </div>
    <div class="container px-0">
      <nav class="navbar navbar-expand-lg navbar-light px-0 py-3">
        <a class="navbar-brand" href="#">
          <img src="../../assets/images/logo.svg" alt="Logo" class="img-responsive">
        </a>
        <button
          class="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse justify-content-end" id="navbarSupportedContent">
          <ul class="navbar-nav mx-auto">
            <li class="nav-item active">
              <a class="nav-link" href="#">
                Why Saleswhale?
                <span class="sr-only">(current)</span>
              </a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="#">Product</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="#">Company</a>
            </li>
          </ul>
          <ul class="navbar-nav navbar-nav-mobile">
            <li class="nav-item">
              <a class="nav-link nav-link-w-100 btn btn-outline-primary" href="#">
                Login
              </a>
            </li>
            <li class="nav-item active">
              <a class="nav-link nav-link-w-100 btn btn-primary ml-3 text-white mobile-margin-0" href="#">
                Get a Demo
              </a>
            </li>
          </ul>
        </div>
      </nav>

      <div class="row mt-5 mobile-padding-0-2 mobile-mt-5">
        <div class="col-lg-12">
          <img src="../../assets/images/banner-version.svg" alt="Header bg" class="w-100 desktop-only">
          <img src="../../assets/images/mobile-banner-version.svg" alt="Header bg" class="w-100 mobile-only">
        </div>
        <div class="col-12 col-md-6 col-lg-6 mt-5 mobile-mt-4">
          <h1 class="wow-h1 animated fadeInUp">A future where AI and humans work hand in hand</h1>
        </div>
        <div class="col-12 col-md-6 col-lg-6 mt-5 px-0 mobile-mt-4">
          <p class="wow-p animated fadeInUp">At Saleswhale, we believe that the future of work is Collaborative Intelligence - where machines and AI help humans fulfil their highest potential by automating drudgery and producing for more effective results.</p>
          <p class="wow-p animated fadeInUp">AI assistants help to automate repetitive and mundane tasks so that us humans focus on creative, high-impact work that creates value and inspires. The result? Happier and more productive teams doing meaningful work.</p>
          <p class="wow-p animated fadeInUp">Our mission is to put an AI assistant in every company that could greatly benefit from it, and we're starting off strong addressing this need in marketing and sales teams all across the world.</p>
          <p class="wow-p animated fadeInUp">The future is where AI and humans work hand in hand, and weâ€™re here to help businesses benefit from that.</p>
        </div>
      </div>
    </div>
  </section>
</template>

<script>
export default {}
</script>

<style lang="scss" scoped>
@media (min-width: 320px) and (max-width: 600px) {
  .navbar-expand-lg {
    margin-left: 15px;
    margin-right: 15px;
  }
  .mobile-margin-0 {
    margin-left: 0!important;
    margin-top: 1rem!important;
  }
  .navbar-nav-mobile {
    display: inline;
    margin-left: 1.5rem;
    li {
      display: inline-block;
    }
    li.active {
      margin-left: .5rem;
    }
  }
}
#header {
  .navbar-brand {
    img {
      height: auto;
      width: 100%;
      max-width: 180px;
    }
  }
  .contact-us {
    background: #000;
    text-align: center;
    color: #fff;
    font-size: 13px;
    padding: 0.5rem 0;
    font-weight: 100;
    a {
      color: #fff;
      text-decoration: underline;
      padding-left: 0.7rem;
    }
  }
  .wow-h1 {
    font-size: 58px;
    line-height: 62px;
    font-weight: bold;
    color: #002240;
    width: 498px;
  }
  .wow-p {
    font-size: 18px;
    color: #002240;
  }
  .navbar-collapse {
    .nav-link-w-100 {
      font-weight: 400 !important;
    }
    .nav-link {
      padding: 0 1.5rem;
      font-style: normal;
      font-weight: bold;
      font-size: 18px;
      line-height: 2.5rem;
      color: #002240;
      i {
        font-size: 1.4rem;
      }
    }
    .header-search {
      border: 1px solid #ebebeb;
      border-radius: 2px;
    }
  }
}
</style>
