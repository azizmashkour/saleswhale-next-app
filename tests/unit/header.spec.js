import { shallowMount } from '@vue/test-utils'
import Header from '@/components/shared/Header'

describe('Header.vue', () => {
  const items = [
    {
      title: 'Why Saleswhale?',
      className: 'sales'
    },
    {
      title: 'Product',
      className: 'product'
    },
    {
      title: 'Company',
      className: 'company'
    }
  ]

  const wrapper = shallowMount(Header, {
    propsData: { items }
  })

  describe('contains navbar menu element ', () => {
    it('renders Header component when mounted', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    it('renders the correct markup', () => {
      expect(wrapper.html()).toContain('<a href="#">OUR COVID-19 RESPONSE</a>')
    })

    it('renders li for each item in props.list', () => {
      expect(wrapper.findAll('li.list-items')).toHaveLength(items.length)
    })

    it('has the login button with the text `Login`', () => {
      expect(wrapper.contains('a.login-btn')).toBe(true)
      expect(wrapper.find('a.login-btn').text()).toMatch('Login')
    })

    it('has the get a demo button with the text `Get a Demo`', () => {
      expect(wrapper.contains('a.get-demo-btn')).toBe(true)
      expect(wrapper.find('a.get-demo-btn').text()).toMatch('Get a Demo')
    })
  })

  describe('contains title and paragraphes ', () => {
    it('renders the header h1 tag title', () => {
      expect(wrapper.contains('h1.wow-h1')).toBe(true)
      expect(wrapper.find('h1.wow-h1').text()).toEqual('A future where AI and humans work hand in hand')
    })

    it('renders the header p tag with wow-p.p1 className', () => {
      expect(wrapper.contains('p.wow-p.p1')).toBe(true)
      expect(wrapper.find('p.wow-p.p1').text()).toEqual('At Saleswhale, we believe that the future of work is Collaborative Intelligence - where machines and AI help humans fulfil their highest potential by automating drudgery and producing for more effective results.')
    })

    it('renders the header p tag with wow-p.p2 className', () => {
      expect(wrapper.contains('p.wow-p.p2')).toBe(true)
      expect(wrapper.find('p.wow-p.p2').text()).toEqual('AI assistants help to automate repetitive and mundane tasks so that us humans focus on creative, high-impact work that creates value and inspires. The result? Happier and more productive teams doing meaningful work.')
    })

    it('renders the header p tag with wow-p.p3 className', () => {
      expect(wrapper.contains('p.wow-p.p3')).toBe(true)
      expect(wrapper.find('p.wow-p.p3').text()).toEqual("Our mission is to put an AI assistant in every company that could greatly benefit from it, and we're starting off strong addressing this need in marketing and sales teams all across the world.")
    })

    it('renders the header p tag with wow-p.p4 className', () => {
      expect(wrapper.contains('p.wow-p.p4')).toBe(true)
      expect(wrapper.find('p.wow-p.p4').text()).toEqual('The future is where AI and humans work hand in hand, and weâ€™re here to help businesses benefit from that.')
    })
  })
})
