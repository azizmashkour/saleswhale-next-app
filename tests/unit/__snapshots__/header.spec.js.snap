// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue contains navbar menu element  renders Header component when mounted 1`] = `
<section
  id="header"
  items="[object Object],[object Object],[object Object]"
>
  <div
    class="contact-us"
  >
    <a
      href="#"
    >
      OUR COVID-19 RESPONSE
    </a>
  </div>
   
  <div
    class="container px-0"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light px-0 py-3"
    >
      <a
        class="navbar-brand"
        href="#"
      >
        <img
          alt="Logo"
          class="img-responsive"
          src="../../assets/images/logo.svg"
        />
      </a>
       
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-target="#navbarSupportedContent"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
       
      <div
        class="collapse navbar-collapse justify-content-end"
        id="navbarSupportedContent"
      >
        <ul
          class="navbar-nav mx-auto"
        >
          <li
            class="nav-item list-items"
          >
            <a
              class="nav-link sales"
              href="#"
            >
              Why Saleswhale?
            </a>
          </li>
          <li
            class="nav-item list-items"
          >
            <a
              class="nav-link product"
              href="#"
            >
              Product
            </a>
          </li>
          <li
            class="nav-item list-items"
          >
            <a
              class="nav-link company"
              href="#"
            >
              Company
            </a>
          </li>
        </ul>
         
        <ul
          class="navbar-nav navbar-nav-mobile"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-w-100 btn btn-outline-primary login-btn"
              href="#"
            >
              
              Login
            
            </a>
          </li>
           
          <li
            class="nav-item active"
          >
            <a
              class="nav-link nav-link-w-100 btn btn-primary ml-3 text-white mobile-margin-0 get-demo-btn"
              href="#"
            >
              
              Get a Demo
            
            </a>
          </li>
        </ul>
      </div>
    </nav>
     
    <div
      class="row mt-5 mobile-padding-0-2 mobile-mt-5"
    >
      <div
        class="col-lg-12"
      >
        <img
          alt="Header bg"
          class="w-100 desktop-only"
          src="../../assets/images/banner-version.svg"
        />
         
        <img
          alt="Header bg"
          class="w-100 mobile-only"
          src="../../assets/images/mobile-banner-version.svg"
        />
      </div>
       
      <div
        class="col-12 col-md-6 col-lg-6 mt-5 mobile-mt-4"
      >
        <h1
          class="wow-h1 animated fadeInUp"
        >
          A future where AI and humans work hand in hand
        </h1>
      </div>
       
      <div
        class="col-12 col-md-6 col-lg-6 mt-5 px-0 mobile-mt-4"
      >
        <p
          class="wow-p p1 animated fadeInUp"
        >
          At Saleswhale, we believe that the future of work is Collaborative Intelligence - where machines and AI help humans fulfil their highest potential by automating drudgery and producing for more effective results.
        </p>
         
        <p
          class="wow-p p2 animated fadeInUp"
        >
          AI assistants help to automate repetitive and mundane tasks so that us humans focus on creative, high-impact work that creates value and inspires. The result? Happier and more productive teams doing meaningful work.
        </p>
         
        <p
          class="wow-p p3 animated fadeInUp"
        >
          Our mission is to put an AI assistant in every company that could greatly benefit from it, and we're starting off strong addressing this need in marketing and sales teams all across the world.
        </p>
         
        <p
          class="wow-p p4 animated fadeInUp"
        >
          The future is where AI and humans work hand in hand, and weâ€™re here to help businesses benefit from that.
        </p>
      </div>
    </div>
  </div>
</section>
`;
